cl/search/views.py

```diff
@@ -41,6 +42,54 @@ def check_pagination_depth(page_number):
     return False
 
 
+def get_mlt_query(cd, facet, solr_url, seed_pk, filter_query):
+    """
+    By default Solr MoreLikeThis queries do not support highlighting. Thus, we use a special search interface
+    and build the Solr query manually.
+
+    :param cd: Cleaned search form data
+    :param facet: Set to True to enable facets
+    :param solr_url: URL from Solr endpoint, e.g. settings.SOLR_OPINION_URL
+    :param seed_pk: ID of the document for that related documents should be returned
+    :param filter_query:
+    :return: Executed SolrSearch
+    """
+    si = MoreLikeThisHighlightsSolrInterface(solr_url, mode='r')
+
+    print('Related for #{}, filter query: {}'.format(seed_pk, filter_query))
+
+    # Reset query for query builder
+    cd['q'] = ''
+
+    # Build main query as always
+    q = build_main_query(cd, facet=facet)
+
+    # Option A: strip field queries, e.g. foo:bar would produce an invalid filter_query
+    # Option B: parse original query + validate all field names (not in use)
+    cleaned_fq = re.sub('[^-!"()0-9a-zA-Z]+', ' ', filter_query.strip())
```

Comment:
I'm not sure I understand what's happening here, but it looks like you're stripping any punctuation that comes after the `related:opinionN` part of the query, and it sounds like the goal is to ensure that people don't have invalid queries? I think it's better to let them make such queries, and if they make mistakes just say there's an error. I think that's how things currently work?

Reply:
Without any validation or cleaning, sending a user input directly as filter query (fq) can cause Solr errors. For instance, `related:123 foo:bar` would throw an error because the field `foo` does not exist.

Reply:
That's *fine* and happens a lot already. When that happens, the front end just throws an error (these could be improved, but I think it's a fine solution for now).

Reply:
That's *fine* and happens a lot already. When that happens, the front end just throws an error (these could be improved, but I think it's a fine solution for now).